### PR TITLE
WPCS : Fix the PHP files in the root directory

### PR DESCRIPTION
**Pull request by Ismail-elkorchi**
While this PR doesn't cover the whole code base nor all the rules it's a first step to start following the WordPress coding standards ( https://github.com/WordPress/theme-check/issues/245 ). 

The first commit changes the line ending of two files so it match the other files. The rest deals with the PHP files that are in the root directory, each commit containing the changes related to one file.

### CODE REVIEW

File: main.php [hunk 1]
```diff
@@ -167,9 +180,11 @@ function tc_form() {
 		echo ( basename( STYLESHEETPATH ) === $location['Stylesheet'] ) ? 'value="' . $location['Stylesheet'] . '" style="font-weight:bold;">' . $name . '</option>' : 'value="' . $location['Stylesheet'] . '">' . $name . '</option>';
 	}
 	echo '</select>';
-	echo '<input class="button" type="submit" value="' . __( 'Check it!', 'theme-check' ) . '" />';
-	if ( defined( 'TC_PRE' ) || defined( 'TC_POST' ) ) echo ' <input name="trac" type="checkbox" /> ' . __( 'Output in Trac format.', 'theme-check' );
-	echo '<input name="s_info" type="checkbox" /> ' . __( 'Suppress INFO.', 'theme-check' );
+	echo '<input class="button" type="submit" value="' . esc_html__( 'Check it!', 'theme-check' ) . '" />';
```
Comment (carolinan) [inline main.php:183]: This would need be esc_attr, since it's inside an attribute.

File: main.php [hunk 2]
```diff
@@ -2,153 +2,166 @@
...
 	<?php
 }
 
 function tc_success() {
 	?>
-	<div class="tc-success"><p><?php _e( 'Now that your theme has passed the basic tests you need to check it properly using the test data before you upload it to the WordPress Themes Directory.', 'theme-check' ); ?></p>
-	<p><?php _e( 'Make sure to review the guidelines at <a href="https://make.wordpress.org/themes/handbook/review/required/">Theme Review</a> before uploading a Theme.', 'theme-check' ); ?></p>
-	<h3><?php _e( 'Useful Links', 'theme-check' ); ?></h3>
+	<div class="tc-success"><p><?php esc_html_e( 'Now that your theme has passed the basic tests you need to check it properly using the test data before you upload it to the WordPress Themes Directory.', 'theme-check' ); ?></p>
+	<p><?php esc_html_e( 'Make sure to review the guidelines at <a href="https://make.wordpress.org/themes/handbook/review/required/">Theme Review</a> before uploading a Theme.', 'theme-check' ); ?></p>
```
Comment (carolinan) [inline main.php:159]: Using esc_html_e here breaks the HTML for the link

File: main.php [hunk 3]
```diff
@@ -2,153 +2,166 @@
...
 		'<a href="https://profiles.wordpress.org/pross/">Pross</a>'
-		); ?></p>
-	<p><?php printf( __( 'If you have found a bug or would like to make a suggestion or contribution, please leave a post on the <a href="%1$s">WordPress forums</a>, or talk about it with the theme review team on <a href="%2$s">Make WordPress Themes</a> site.', 'theme-check' ), 'https://wordpress.org/tags/theme-check?forum_id=10', 'https://make.wordpress.org/themes/') ; ?></p>
-	<p><?php printf( __( 'The code for Theme Check can be contributed to on <a href="%s">GitHub</a>.', 'theme-check' ), 'https://github.com/Otto42/theme-check'); ?></p>
-	<h3><?php _e( 'Testers', 'theme-check' ); ?></h3>
-	<p><a href="https://make.wordpress.org/themes/"><?php _e( 'The WordPress Theme Review Team', 'theme-check' ); ?></a></p>
+	);
+	?>
+		</p>
+	<p><?php printf( esc_html__( 'If you have found a bug or would like to make a suggestion or contribution, please leave a post on the <a href="%1$s">WordPress forums</a>, or talk about it with the theme review team on <a href="%2$s">Make WordPress Themes</a> site.', 'theme-check' ), 'https://wordpress.org/tags/theme-check?forum_id=10', 'https://make.wordpress.org/themes/' ); ?></p>
```
Comment (carolinan) [inline main.php:149]: Same here, esc_html breaks the output of the links.

File: main.php [hunk 4]
```diff
@@ -2,153 +2,166 @@
...
+	<p><?php printf( esc_html__( 'The code for Theme Check can be contributed to on <a href="%s">GitHub</a>.', 'theme-check' ), 'https://github.com/Otto42/theme-check' ); ?></p>
```
Comment (carolinan) [inline main.php:150]: Note to self :) this github link needs to be updated.

File: main.php [hunk 5]
```diff
@@ -2,153 +2,166 @@
...
 			tc_success();
 		}
-		if ( !defined( 'WP_DEBUG' ) || WP_DEBUG == false ) echo '<div class="updated"><span class="tc-fail">' . __('WARNING','theme-check') . '</span> ' . __( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
+		if ( ! defined( 'WP_DEBUG' ) || WP_DEBUG === false ) {
+			echo '<div class="updated"><span class="tc-fail">' . esc_html__( 'WARNING', 'theme-check' ) . '</span> ' . esc_html__( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
+		}
 		echo '<div class="tc-box">';
 		echo '<ul class="tc-result">';
-		echo $results;
+		echo esc_html( $results );
```
Comment (carolinan) [inline main.php:98]: Using esc_html here breaks the output of the results which makes use of lists etc.

File: main.php [hunk 6]
```diff
@@ -2,153 +2,166 @@
...
+		if ( ! $success ) {
+			echo '<h2>' . sprintf( esc_html__( 'One or more errors were found for %1$s.', 'theme-check' ), $data['Title'] ) . '</h2>';
 		} else {
-			echo '<h2>' . sprintf(__('%1$s passed the tests', 'theme-check'), $data[ 'Title' ] ) . '</h2>';
+			echo '<h2>' . sprintf( __( '%1$s passed the tests', 'theme-check' ), $data['Title'] ) . '</h2>';
 			tc_success();
 		}
-		if ( !defined( 'WP_DEBUG' ) || WP_DEBUG == false ) echo '<div class="updated"><span class="tc-fail">' . __('WARNING','theme-check') . '</span> ' . __( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
+		if ( ! defined( 'WP_DEBUG' ) || WP_DEBUG === false ) {
+			echo '<div class="updated"><span class="tc-fail">' . esc_html__( 'WARNING', 'theme-check' ) . '</span> ' . esc_html__( '<strong>WP_DEBUG is not enabled!</strong> Please test your theme with <a href="https://wordpress.org/support/article/editing-wp-config-php/">debug enabled</a> before you upload!', 'theme-check' ) . '</div>';
```
Comment (carolinan) [inline main.php:94]: This breaks the output of the HTML in the text...

File: main.php (hunk 1 above)
Comment (pattonwebz) [reply main.php:183]:
```suggestion
	echo '<input class="button" type="submit" value="' . esc_attr__( 'Check it!', 'theme-check' ) . '" />';
```

File: main.php (hunk 2 above)
Comment (pattonwebz) [reply main.php:159]:
```suggestion
	<p>
		<?php
		printf(
			/* translators: 1 is an opening anchor tag 2 is the closing part of the tag. */
			esc_html( 'Make sure to review the guidelines at %1$sTheme Review%2$s before uploading a Theme.', 'theme-check' ),
			'<a href="https://make.wordpress.org/themes/handbook/review/required/">',
			'</a>'
		);
		?>
	</p>
```

**Comment by Ismail-elkorchi** [comment]
Thank you @carolinan for the review, and sorry for the late reply. I have processed your feedback, and pushed some more changes which should be easy to review.

**Comment by dingo-d** [comment]
Did you sync with the master branch? We did a release yesterday and merged some changes in the past 2 days.

**Comment by Ismail-elkorchi** [comment]
@dingo-d Yes ! All the new changes that were merged into master are now part of this PR.

**Comment by Ismail-elkorchi** [comment]
Is there anything I can do to help move this PR forward?

**Comment by Ismail-elkorchi** [comment]
@carolinan, @dingo-d, @pattonwebz just wanted to check in with this—is the PR ready to merge?

**Comment by pattonwebz** [comment]
@Ismail-elkorchi I'll look it over today. Thanks for doing this :)

**Comment by Ismail-elkorchi** [comment]
@pattonwebz Thanks. I'm looking forward to send more PRs.

**Comment by Ismail-elkorchi** [comment]
@pattonwebz Have you had time to look into this?

**Comment by pattonwebz** [comment]
@Ismail-elkorchi sorry this PR was bigger than I first expected but nice job on this and thank you for taking care of all the CS issues in those files.

All looks good to me so approved the PR.

Again thank you for this and I will see about merging it soon and rolling out a release with a few small tweaks and fixes :)

**Comment by Ismail-elkorchi** [comment]
Hi @pattonwebz ! Thank you for your time and for your kind words. No need to be sorry, I admit that this PR is not the simplist to review.

To make the review easier in futur PR's, I think of enabling Travis, and disabling all the WPCS rules, then in each PR we can enable a rule, and fix all the errors related to it. I am curious to hear your thought on this.

File: main.php [hunk 7]
```diff
@@ -2,153 +2,198 @@
...
+		}
 		echo '<div class="tc-box">';
 		echo '<ul class="tc-result">';
-		echo $results;
+		echo wp_kses(
+			$results,
+			array(
+				'li' => array(),
+			)
+		);
```
[the remark this answers is not in the document]
Comment (pattonwebz) [reply main.php]: Added a commit that solves this

**Comment by pattonwebz** [comment]
Hey @Ismail-elkorchi I finished testing this and all looks good. Nice work on this :)

The only thing I spotted was that you had a wp_kses() that was not allowing strong and span tags though. I pushed some commits into your branch before merging to handle that.

**Comment by Ismail-elkorchi** [comment]
Hey @pattonwebz ! Thank you very much for your careful review, and for the changes you made. Thank you @carolinan for reviewing the changes and pointing out the errors I made. I was very delighted to see that my criticism was heard taken very seriously. And lastly, thanks to all the members of the Themes Team for their great and hard work.

**Pull request merged**